### PR TITLE
wishlist recyclerview under profile

### PR DESCRIPTION
**Pull request by Alvtag**
**Description**

**Issues that should be CLOSED by merge of this PR:**
* Fixes #
TPP267

### CODE REVIEW

File: traveler_ui_kit/src/main/java/com/guestlogix/traveleruikit/activities/CatalogItemDetailsActivity.java [hunk 1]
```diff
@@ -16,17 +17,16 @@
...
+import com.guestlogix.traveleruikit.fragments.*;
 import com.guestlogix.traveleruikit.utils.FragmentTransactionQueue;
 import com.guestlogix.traveleruikit.viewmodels.CatalogItemDetailsViewModel;
 import com.guestlogix.traveleruikit.viewmodels.StatefulViewModel;
 
 public class CatalogItemDetailsActivity extends AppCompatActivity implements CatalogItemDetailsCallback, RetryFragment.InteractionListener {
 
     public static final String ARG_PRODUCT = "product";
+    public static final String KEY_PRODUCT_WISHLIST_STATE = "key_product_wishlist_state";
+    public static final String KEY_PRODUCT= "key_product";
```
[the remark this answers is not in the document]
Comment (Alvtag) [reply traveler_ui_kit/src/main/java/com/guestlogix/traveleruikit/activities/CatalogItemDetailsActivity.java]: removed as per other comment

File: traveler_ui_kit/src/main/java/com/guestlogix/traveleruikit/fragments/CatalogItemDetailsFragment.java [hunk 2]
```diff
@@ -161,6 +165,10 @@ public void onClick(View v) {
     @Override
     public void onWishlistAddSuccess(Product item, CatalogItemDetails itemDetails) {
         catalogItemDetails = itemDetails;
+        Activity activity = getActivity();
```
Comment (ata-n) [inline traveler_ui_kit/src/main/java/com/guestlogix/traveleruikit/fragments/CatalogItemDetailsFragment.java]: Fragments should not know of the activity that may or may not host them

Comment (ata-n) [reply traveler_ui_kit/src/main/java/com/guestlogix/traveleruikit/fragments/CatalogItemDetailsFragment.java]: I think a different pattern is used for this type of communication

File: traveler_app/src/main/res/layout/activity_profile.xml [hunk 3]
```diff
@@ -2,104 +2,128 @@
...
+    </LinearLayout>
+
+    <LinearLayout
+            android:id="@+id/linearlayout_email"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:orientation="horizontal">
+
+        <ImageView
+                android:id="@+id/imageView_profileItem_icon"
```
Comment (amirziarati) [inline traveler_app/src/main/res/layout/activity_profile.xml]: dont you have a convention on vies' names ? the best one is the same name used in java code and also starting with some kinda initial for the type of view. (e.g.: tvPassword, btnSend, imgProfile)

same naming in java and xml helps to have more readable code and migrating to kotlin would be easier (in kotlin views in xml and java code has the same name)

File: traveler_core_kit/src/androidTest/java/com/guestlogix/travelercorekit/TravelerTest.java [hunk 4]
```diff
@@ -317,7 +317,7 @@ public void onCatalogSuccess(Catalog catalog) {
                 Assert.assertNotNull(product);
                 productIdReference.reference = product.getId();
 
-                Traveler.fetchWishlist(new WishlistQuery(0, 10, new Date(), new Date()), 0, step3WishlistFetchCallback);
+                Traveler.fetchWishlist(new WishlistQuery(0, 1000, new Date(), new Date()), 0, step3WishlistFetchCallback);
```
Comment (amirziarati) [inline traveler_core_kit/src/androidTest/java/com/guestlogix/travelercorekit/TravelerTest.java]: magic numbers makes code hard to understand especially for a new comer. even a local constant with a well defined name helps a lot

File: traveler_ui_kit/src/main/java/com/guestlogix/traveleruikit/fragments/CatalogItemDetailsFragment.java [hunk 5]
```diff
@@ -178,6 +186,10 @@ public void onWishlistAddError(Error error) {
     @Override
     public void onWishlistRemoveSuccess(Product item, CatalogItemDetails itemDetails) {
         catalogItemDetails = itemDetails;
+        Activity activity = getActivity();
+        if (activity instanceof CatalogItemDetailsActivity) {
```
Comment (amirziarati) [inline traveler_ui_kit/src/main/java/com/guestlogix/traveleruikit/fragments/CatalogItemDetailsFragment.java]: same here. this shows something wrong in the way things are done

File: traveler_ui_kit/src/main/java/com/guestlogix/traveleruikit/fragments/CatalogItemDetailsFragment.java [hunk 6]
```diff
@@ -161,6 +165,10 @@ public void onClick(View v) {
     @Override
     public void onWishlistAddSuccess(Product item, CatalogItemDetails itemDetails) {
         catalogItemDetails = itemDetails;
+        Activity activity = getActivity();
+        if (activity instanceof CatalogItemDetailsActivity) {
```
[the remark this answers is not in the document]
Comment (Alvtag) [reply traveler_ui_kit/src/main/java/com/guestlogix/traveleruikit/fragments/CatalogItemDetailsFragment.java]: updated

File: traveler_ui_kit/src/main/java/com/guestlogix/traveleruikit/fragments/CatalogItemDetailsFragment.java (hunk 5 above)
Comment (Alvtag) [reply traveler_ui_kit/src/main/java/com/guestlogix/traveleruikit/fragments/CatalogItemDetailsFragment.java]: updated

File: traveler_core_kit/src/androidTest/java/com/guestlogix/travelercorekit/TravelerTest.java (hunk 4 above)
Comment (Alvtag) [reply traveler_core_kit/src/androidTest/java/com/guestlogix/travelercorekit/TravelerTest.java]: updated

**Pull request merged**